src/components/shared/Entity/ExistingEntityCards/Cards/Existing.tsx

```diff
@@ -0,0 +1,175 @@
+import Editor from '@monaco-editor/react';
+import {
+    Box,
+    ButtonBase,
+    Collapse,
+    Paper,
+    Typography,
+    useTheme,
+} from '@mui/material';
+import {
+    CaptureQueryWithSpec,
+    MaterializationQueryWithSpec,
+} from 'api/liveSpecsExt';
+import { authenticatedRoutes } from 'app/routes';
+import ConnectorLogo from 'components/connectors/card/Logo';
+import CustomWidthTooltip from 'components/shared/CustomWidthTooltip';
+import {
+    alternateConnectorImageBackgroundSx,
+    monacoEditorComponentBackground,
+    semiTransparentBackground,
+    semiTransparentBackgroundIntensified,
+} from 'context/Theme';
+import { GlobalSearchParams } from 'hooks/searchParams/useGlobalSearchParams';
+import { isEmpty } from 'lodash';
+import LogRocket from 'logrocket';
+import { useState } from 'react';
+import { useIntl } from 'react-intl';
+import { useNavigate } from 'react-router';
+import { CustomEvents } from 'services/logrocket';
+import { stringifyJSON } from 'services/stringify';
+import { getPathWithParams } from 'utils/misc-utils';
+
+interface Props {
+    queryData: CaptureQueryWithSpec | MaterializationQueryWithSpec | null;
+}
+
+const EDITOR_HEIGHT = 396;
+
+const trackEvent = (
+    logEvent: CustomEvents,
+    data: CaptureQueryWithSpec | MaterializationQueryWithSpec
+) => {
+    LogRocket.track(logEvent, {
+        id: data.id,
+        connector_tag_id: data.connector_id,
+    });
+};
+
+function ExistingEntityCard({ queryData }: Props) {
+    const navigate = useNavigate();
+
+    const theme = useTheme();
+    const intl = useIntl();
+
+    const [detailsExpanded, setDetailsExpanded] = useState<boolean>(false);
+
+    const handlers = {
+        editTask: () => {
+            if (!isEmpty(queryData)) {
+                const logEvent =
+                    queryData.spec_type === 'capture'
+                        ? CustomEvents.CAPTURE_CREATE_CONFIG_EDIT
+                        : CustomEvents.MATERIALIZATION_CREATE_CONFIG_EDIT;
```

Comment:
I feel this logic should live in the `trackEvent` function so it is self contained.